### PR TITLE
Propagate auto_id_timestamp in primary-replica resync

### PR DESCRIPTION
**Pull request by dnhatn**
A follow-up of #33693 to propagate max_seen_auto_id_timestamp in a primary-replica resync.

Relates #33693

### CODE REVIEW

File: server/src/test/java/org/elasticsearch/index/replication/RecoveryDuringReplicationTests.java [hunk 1]
```diff
@@ -633,6 +634,46 @@ public long indexTranslogOperations(final List<Translog.Operation> operations, f
...
+                }
+            }
+            assertThat(replica1.getMaxSeenAutoIdTimestamp(), equalTo(maxTimestampOnReplica1));
+            assertThat(replica2.getMaxSeenAutoIdTimestamp(), equalTo(maxTimestampOnReplica2));
+            shards.promoteReplicaToPrimary(replica1).get();
+            assertThat(replica2.getMaxSeenAutoIdTimestamp(), equalTo(maxTimestampOnReplica1));
+            for (IndexRequest request : replicationRequests) {
+                shards.index(request); // deliver via normal replication
+            }
+        }
```
Comment (bleskes) [inline server/src/test/java/org/elasticsearch/index/replication/RecoveryDuringReplicationTests.java]: I think we miss an assertion to check that all shards are identical?

File: server/src/test/java/org/elasticsearch/index/shard/PrimaryReplicaSyncerTests.java [hunk 2]
```diff
@@ -76,7 +76,7 @@ public void testSyncerSendsOffCorrectDocuments() throws Exception {
             // Index doc but not advance local checkpoint.
             shard.applyIndexOperationOnPrimary(Versions.MATCH_ANY, VersionType.INTERNAL,
                 SourceToParse.source(shard.shardId().getIndexName(), "_doc", Integer.toString(i), new BytesArray("{}"), XContentType.JSON),
-                IndexRequest.UNSET_AUTO_GENERATED_TIMESTAMP, false);
+                randomNonNegativeLong(), randomBoolean());
```
Comment (bleskes) [inline server/src/test/java/org/elasticsearch/index/shard/PrimaryReplicaSyncerTests.java]: I think we randomly want to have normal ops here (i.e. UNSET_AUTO_GENERATED_TIMESTAMP)

File: server/src/main/java/org/elasticsearch/action/resync/ResyncReplicationRequest.java [hunk 3]
```diff
@@ -73,6 +80,11 @@ public void readFrom(final StreamInput in) throws IOException {
         } else {
             trimAboveSeqNo = SequenceNumbers.UNASSIGNED_SEQ_NO;
         }
+        if (in.getVersion().onOrAfter(Version.V_7_0_0_alpha1)) {
+            maxSeenAutoIdTimestampOnPrimary = in.readZLong();
+        } else {
+            maxSeenAutoIdTimestampOnPrimary = -1;
```
Comment (bleskes) [inline server/src/main/java/org/elasticsearch/action/resync/ResyncReplicationRequest.java]: can we use `UNSET_AUTO_GENERATED_TIMESTAMP`?

**Review by bleskes — APPROVED** [review]
LGTM

File: server/src/test/java/org/elasticsearch/index/replication/RecoveryDuringReplicationTests.java (hunk 1 above)
Comment (dnhatn) [reply server/src/test/java/org/elasticsearch/index/replication/RecoveryDuringReplicationTests.java]: We have this check in the teardown but yes, I added an explicit one here :).

**Comment by dnhatn** [comment]
Thanks @bleskes.

**Pull request merged**